jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -726,23 +739,24 @@ def _update_stretch_histogram(self, msg={}):
 
         # update the n_bins since this may be a new layer
         self._histogram_nbins_changed()
+        # update the curve/colorbar
+        self._update_stretch_curve(msg)
 
-    @observe('is_active', 'image_color_mode_value', 'image_color_value', 'image_colormap_value',
+    @observe('image_color_mode_value', 'image_color_value', 'image_colormap_value',
              'image_contrast_value', 'image_bias_value',
+             'stretch_hist_nbins',
+             'stretch_curve_visible',
              'stretch_function_value', 'stretch_vmin_value', 'stretch_vmax_value',
-             'stretch_hist_nbins', 'stretch_hist_zoom_limits')

```

Comment:
Are you sure we don't need `'stretch_hist_zoom_limits'`?

Reply:
`'is_active', 'layer_selected', 'viewer_selected', 'stretch_hist_zoom_limits'` all trigger `_update_stretch_histogram`, which in turn will call this after updating the histogram.